docs/debugger/sessions.rst

```diff
@@ -120,29 +127,135 @@ that is the situation, then the Makefile will get recreated and you
+Again, I'll invoke the debugger initially.
+
+.. code:: console
+
+    $ remake -X
+    Reading makefiles...
+    Updating makefiles...
+    -> (/tmp/libcdio-paranoia/Makefile:428)
+    Makefile: Makefile.in config.status
+    remake<0>
+
+Instead of stepping we can set a breakpoint on the `dist` target and
+continue running to that point in one command, using :ref:`continue <continue>`.
+
+.. code:: console
+
+    remake<0> continue dist
+    Breakpoint 1 on target `dist', mask 0x0f: file Makefile, line 703.
+    Updating goal targets...
+    -> (/src/external-vcs/github/rocky/libcdio-paranoia/Makefile:703)
+    dist:
+    remake<1>
+
+Now when I issue a `step`, I will step into the commands associated with the `dist` target:
+
+.. code:: console
+
+   remake<1> step
+   File 'dist' does not exist.
+   Must remake target 'dist'.
+   Makefile:704: target 'dist' does not exist
+   ##>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
+   remake  dist-bzip2 dist-gzip am__post_remove_distdir='@:'
+   ##<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
+   ++ (/src/external-vcs/github/rocky/libcdio-paranoia/Makefile:703)
+   dist
+   remake<2>
+
+Notice that the event icon above is `++` which means I am stepping shell commands, here those associated with the Make target `dist`.
+Above the line with the event icon in between the two chevrons is the command that is *about* to be run.
 
-*to be continued...*
 
 Debugging Make Variables
 -------------------------
 
-*to be continued...*
+In the above session we have seen that output has variables expanded. You can query any GNU Make variable
+that has been set in the program without variables inside expanded using the :ref:`print <print>` command.
 
+.. code:: console
+
+    remake<2> print MAKE
+    (origin default) MAKE = $(MAKE_COMMAND)
+
+The `(origin default)` means this is a built-in definition. The other
+kind of print which does full expansion of the variables is called
+`expand` or `x`. Here is an example
+
+.. code:: console
+
+    remake<3> expand MAKE
+    (origin default) MAKE := remake
+
+Note that in printing expanded values we use `:=` while non-expanded
+values we use `=` This output matches the semantics of these
+assignment operators.
+
+In fact, `expand` doesn't need a variable name, it will work with a
+string. So I could type `x This is $(MAKE)` or `x $(MAKE) $(DIST_TARGETS)`
+For the latter, I get:
+
+.. code:: console
+
+    remake<4> x $(MAKE) $(DIST_TARGETS)
+    remake dist-bzip2 dist-gzip
+
+No location identification is given here since what I put in isn't a
+variable. Also note that for `expand` I add the dollar sign and
+parenthesis when there is other stuff. If you just want information
+about the variable you can leave that off.
+
+However for `print` you *never* add the dollar sign; printing only
+prints *variables* not strings.
+
+I change values too using either the :ref:`set <set>`, :ref:`set <setq>` or
```

Comment:
"I change values..." > "I can changes values..." ?

Reply:
Yes, again you are correct. I make a *lot* of mistakes which is one reason debuggers are so dear to me. Should be fixed in 80681f8